pkg/armhelpers/storage.go

```diff
@@ -43,9 +44,29 @@ func (az *AzureClient) getStorageKeys(ctx context.Context, resourceGroup, accoun
 // TODO(colemick): why doesn't SDK give a way to just delete a blob by URI?
 // it's what it ends up doing internally anyway...
 func (as *AzureStorageClient) DeleteBlob(vhdContainer, vhdBlob string) error {
-	bs := as.client.GetBlobService()
-	containerRef := bs.GetContainerReference(vhdContainer)
+	containerRef := getContainerRef(as, vhdContainer)
 	blobRef := containerRef.GetBlobReference(vhdBlob)
 
 	return blobRef.Delete(&azStorage.DeleteBlobOptions{})
 }
+
+// CreateContainer creates the CloudBlobContainer if it does not exist
+func (as *AzureStorageClient) CreateContainer(containerName string) (bool, error) {
+	containerRef := getContainerRef(as, containerName)
+	created, err := containerRef.CreateIfNotExists(nil)
+
+	return created, err
+}
+
+// SaveBlockBlob initializes a block blob by taking the byte
+func (as *AzureStorageClient) SaveBlockBlob(containerName, blobName string, b []byte, options *azStorage.PutBlobOptions) error {
```

Comment:
Instead of passing an azStorage options as a parameter and necessitating an import in other packages, we can add an options field in the AzureStorageClient and then construct the PutBlobOptions from that. Better SOC that way IMO.

Reply:
I see that PutBlobOptions is composed of headers. So the new options field in AzureStorageClient can be a Map.

Reply:
If we put the option to the AzureStorageClient field, the values of the option will be predefined and we lose the agility of changing the option when saving a specific blob with a specific option.
We may set this field to a public but that also require the caller to setthe option values everytime it calls the SaveBlockBlob.

Reply:
I see your point. Then we should not be using an AzureStorageClient wrapper in the first place.